src/Evolution/Systems/CurvedScalarWave/Worldtube/SingletonActions/IterateAccelerationTerms.cpp

```diff
@@ -25,20 +26,183 @@ void IterateAccelerationTerms::apply(
+        second_spatial_derivative_inverse_ks_metric(particle_position);
+    const auto di_trace_christoffel =
+        spatial_derivative_ks_contracted_christoffel(particle_position);
+    const auto di_metric = spatial_derivative_ks_metric(metric, di_imetric);
+    const auto dij_metric = second_spatial_derivative_metric(
+        metric, di_metric, di_imetric, dij_imetric);
+    const auto di_christoffel = spatial_derivative_christoffel(
+        di_metric, dij_metric, imetric, di_imetric);
+    const auto& u0 = get(get<Tags::TimeDilationFactor>(background));
+
+    // The evolution of the mass is a second order effect so we only include it
+    // starting at the 2nd iteration.
+    const double evolved_mass = iteration == 1
+                                    ? mass.value()
+                                    : mass.value() - charge * get(psi_monopole);
+
+    const auto self_force_acc = self_force_acceleration(
+        dt_psi_monopole, psi_dipole, particle_velocity, charge, evolved_mass,
+        imetric, get<Tags::TimeDilationFactor>(background));
+
+    const double t_minus_turnup = time - turn_on_time.value();
+    roll_on = turn_on_function(t_minus_turnup, turn_on_interval.value());
+    const double dt_roll_on =
+        dt_turn_on_function(t_minus_turnup, turn_on_interval.value());
+    const double dt2_roll_on =
+        dt2_turn_on_function(t_minus_turnup, turn_on_interval.value());
+    for (size_t i = 0; i < Dim; ++i) {
+      acc.get(i) += roll_on * self_force_acc.get(i);
+    }
+    const tnsr::A<double, Dim> u{{u0, get<0>(particle_velocity) * u0,
+                                  get<1>(particle_velocity) * u0,
+                                  get<2>(particle_velocity) * u0}};
+    const tnsr::a<double, Dim> d_psiR{{get(dt_psi_monopole), get<0>(psi_dipole),
+                                       get<1>(psi_dipole), get<2>(psi_dipole)}};
+
+    // Eq.(55)
+    double dt2_psiR = get<0>(trace_christoffel) * get<0>(d_psiR);
+    for (size_t i = 0; i < Dim; ++i) {
+      dt2_psiR += -2. * imetric.get(0, i + 1) * dt_psi_dipole.get(i) +
+                  trace_christoffel.get(i + 1) * psi_dipole.get(i);
+    }
+    dt2_psiR /= get<0, 0>(imetric);
+
+    tnsr::a<double, Dim> dt_d_psiR{{dt2_psiR, get<0>(dt_psi_dipole),
+                                    get<1>(dt_psi_dipole),
+                                    get<2>(dt_psi_dipole)}};
+    for (size_t i = 0; i < Dim; ++i) {
+      get<0>(dt_d_psiR) += particle_velocity.get(i) * dt_psi_dipole.get(i);
+    }
+
+    double dt_evolved_mass = get(dt_psi_monopole);
+    double dt2_evolved_mass = dt2_psiR;
+    for (size_t i = 0; i < Dim; ++i) {
+      dt_evolved_mass += psi_dipole.get(i) * particle_velocity.get(i);
+      dt2_evolved_mass += 2. * dt_psi_dipole.get(i) * particle_velocity.get(i) +
+                          psi_dipole.get(i) * acc.get(i);
+    }
+    dt_evolved_mass *= iteration == 1 ? 0. : -charge;
+    dt2_evolved_mass *= iteration == 1 ? 0. : -charge;
```

Comment:
Set the time derivs to zero initially and then only if the iteration isn't 1 do the extra computation.